### ISSUE TITLE
S3 snapshot - Access Denied

### ISSUE THREAD

**Issue by hrvatskibogmars** [rancher/rke#2445]
**RKE version:**

`
Running RKE version: v1.2.5
`
**Docker version: (`docker version`,`docker info` preferred)**

```
Client: Docker Engine - Community
 Version:           20.10.2
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        2291f61
 Built:             Mon Dec 28 16:17:43 2020
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.2
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       8891c58
  Built:            Mon Dec 28 16:15:19 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**

`Linux cn-5.4.0-31-generic #35-Ubuntu SMP Thu May 7 20:20:34 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**

AWS - S3

**cluster.yml file:**
```
kubernetes_version: "v1.19.3-rancher1-1"
cluster_name: cn-dev
bastion_host:
  address: X.X.X.X
  user: root
  port: 22
  ssh_key_path: ~/.ssh/global_ssh_key
authentication:
    strategy: x509
    sans:
      - "X.X.X.X"
nodes:
  - address: X.X.X.X
    internal_address: X.X.X.X
    hostname_override: X-cn-dev-master-1
    user: root
    role: [controlplane, etcd]
    ssh_key_path: ~/.ssh/global_ssh_key
    port: 22

  - address: X.X.X.X
    internal_address: X.X.X.X
    hostname_override: X-cn-dev-master-2
    user: root
    role: [controlplane, etcd]
    ssh_key_path: ~/.ssh/global_ssh_key
    port: 22

  - address: X.X.X.X
    internal_address: X.X.X.X
    hostname_override: X-cn-dev-master-3
    user: root
    role: [controlplane, etcd]
    ssh_key_path: ~/.ssh/global_ssh_key
    port: 22

  - address: X.X.X.X
    internal_address: X.X.X.X
    hostname_override: x-cn-dev-worker-1
    user: root
    role: [worker]
    ssh_key_path: ~/.ssh/global_ssh_key
    port: 22
  - ...
services:
  etcd:
    backup_config:
      enabled: true     # enables recurring etcd snapshots
      interval_hours: 6 # time increment between snapshots
      retention: 60     # time in days before snapshot purge
      # Optional S3
      s3backupconfig:
        access_key: "XXXXX"
        secret_key:  "XXX"
        bucket_name: "bucket-cluster-backup"
        endpoint: "s3.eu-west-1.amazonaws.com"
        region: "eu-west-1"
```

```
{
    "Statement": [
        {
            "Action": "s3:*",
            "Effect": "Allow",
            "Resource": [
                "arn:aws:s3:::{BUCKET_NAME}",
                "arn:aws:s3:::{BUCKET_NAME}/*"
            ]
        },
        {
            "Action": [
                "s3:Get*",
                "s3:List*"
            ],
            "Effect": "Allow",
            "Resource": [
                "arn:aws:s3:*:{ACCOUNT_ID}:accesspoint/*"
            ]
        },
        {
            "Action": [
                "s3:GetAccessPoint",
                "s3:GetAccountPublicAccessBlock",
                "s3:ListAllMyBuckets",
                "s3:ListAccessPoints"
            ],
            "Effect": "Allow",
            "Resource": "*"
        }
    ],
    "Version": "2012-10-17"
}
```

**Steps to Reproduce:**

`rke --debug etcd snapshot-save --name test-1611847953.db --config rancher-cn-dev.yaml
`
**Results:**

```
time="2021-01-28T13:34:10Z" level=info msg="invoking uploading backup file [2021-01-28T13:34:08Z_etcd.zip] to s3"
time="2021-01-28T13:34:14Z" level=info msg="failed to upload etcd snapshot file: Access Denied, retried 0 times"
time="2021-01-28T13:34:19Z" level=info msg="failed to upload etcd snapshot file: Access Denied, retried 1 times"
time="2021-01-28T13:34:24Z" level=info msg="failed to upload etcd snapshot file: Access Denied, retried 2 times"
time="2021-01-28T13:34:28Z" level=info msg="failed to upload etcd snapshot file: Access Denied, retried 3 times"
```

Basically, the snapshot failed with Access Denied I used policies from Rancher docs. Can you advise how to change policies so it works?

**Comment by superseb** [comment]
Please share the exact URL of the page you are using in the docs for reference. Did you try giving all permissions to confirm that it is a limited permission error?

**Comment by hrvatskibogmars** [comment]
@superseb https://rancher.com/docs/rancher/v2.x/en/backups/v2.5/configuration/backup-config/#iam-permissions-for-ec2-nodes-to-access-s3

**Comment by hrvatskibogmars** [comment]
I tried giving it full permissions but it's not working (failed to upload etcd snapshot file: Access Denied)

**Comment by superseb** [comment]
That docs page is not for RKE snapshots, it is for Rancher backups. If you want to use IAM for RKE snapshots, you don't need to specify access key and secret key, so the logic uses IAM. If you are already doing that and the IAM role is attached and it has all permissions, I can't really give you a list of permissions if it doesn't work with all permissions. I would suggest testing S3 with AWS CLI and using the IAM role to verify you can contact S3 so we can use that to debug the code/library we use to contact S3.
